### ISSUE TITLE
 Recurring events are not being picked up by the plugin.

### ISSUE THREAD

**Issue by ramsabarish** [EddyVerbruggen/Calendar-PhoneGap-Plugin#156]
Create an event for today in Google calendar. Set it to repeat everyday for a week. This event will not be picked up by the plugin.

**Comment by Appstractie** [comment]
As most of my user's events are recurring, this is a real bummer to this plugin.

**Comment by EddyVerbruggen** [comment]
Guys, how are you creating that event.. I just tried this on my Android 5.1.1 device again and it worked fine:

```js
  function testIssue156_recurringEvent() {
    var startDate = new Date();
    startDate.setHours(20);
    var endDate = new Date();
    endDate.setHours(21);
    var calOptions = window.plugins.calendar.getCalendarOptions();
    calOptions.recurrence = "weekly";
    calOptions.recurrenceEndDate = new Date(2015,5,14,0,0,0,0,0);
    calOptions.url = "http://www.x-services.nl";
    var title = "Weekly recurring";
    var location = "Home";
    notes = null;
    var success = function (message) { alert("Success: " + JSON.stringify(message)); };
    var error = function (message) { alert("Error: " + message); };
    window.plugins.calendar.createEventWithOptions(title,location,notes,startDate,endDate,calOptions,success,error);
  }
```

**Comment by ramsabarish** [comment]
Well I'm not facing an issue while creating a recurring event, the issue is when I try to find the events for a particular day the plugin doesn't pickup the recurring events. Please help me out.

**Comment by Appstractie** [comment]
Hi Eddy,

Thanks for replying. I have created recurring events on my Android 5 Note 3 in 2 different ways, in both ways, the recurring event is ignored by the plugin:

- Directly entering an event in the Samsung S Calendar
- Active Sync from my office

The funny fact is that when I change the name of a single event in the recurring chain (after I create the recurring event), the meeting is seen by the plugin :).

I do not have any issue creating recurring events btw, only looking them up seems not possible right now.

**Comment by EddyVerbruggen** [comment]
With the current version of the plugin I am not able to reproduce this: creating a weekly and daily recurring event works as expected, as well as finding them again with `listEventsInRange`.

It would be very helpful if you can post a sample project reproducing the problem, or at least some code snippets like I did.

**Comment by ramsabarish** [comment]
var today = new Date();   
var dd = today.getDate();
var mm = today.getMonth();
var yyyy = today.getFullYear();

if (dd < 10) {
    dd = '0' + dd;
}

if (mm < 10) {
    mm = '0' + mm;
}

var Startday = new Date(yyyy, mm, dd , 0,0,0,0,0);

var Next_Day = parseInt(dd) + 1;

if (Next_Day < 10) {
    Next_Day = '0' + Next_Day;
}

var EndDay = new Date(yyyy, mm, Next_Day ,0,0,0,0,0);


mm = parseInt(mm) + 1;
if (mm < 10) {
    mm = '0' + mm;
}  

window.plugins.calendar.listEventsInRange(Startday, EndDay, successToday, error);

**Comment by Appstractie** [comment]
This code is part of an Angular factory, to keep it simple, I only extract the parts where I request the meetings with the listEventsInRange() function. With this code, all meetings show up, except the recurring events. 

1. Request the data

 $ionicPlatform.ready(function () {
                    var o = new Date();
                    var y = o.getFullYear();
                    var m = o.getMonth();
                    var d = o.getDate();
                    var t = d;

                    var startDate = new Date(y,m,d,0,0,0,0,0);
                    var endDate = new Date(y,m,t,0,0,0,0,0);
                window.plugins.calendar.listEventsInRange(startDate,endDate,success,error);
});

2. See the data in the success callback

var success = function (data) {
        console.log(data); // no recurring event
}

3. See that the recurring event is present in the calendar
![picture](https://cloud.githubusercontent.com/assets/12409665/7667459/2e25b920-fc09-11e4-8fb0-a74ccf648e4d.jpeg)

**Comment by Appstractie** [comment]
I got the recurring event to show up, but only after I changed the name of a single event in the recurring chain. When I changed the name back to the original name, it is still there in the success callback. 

I have also tested this by changing the name of the whole recurring chain, but then the event is not visible. Adding extra data to recurring events does not make a difference (description, location, etc).

In short: recurring events are not visible right after creation, only after modifying them one by one. 

I have tested this with all my 4 local and cloud accounts, all with consistent results for recurring events.

**Comment by EddyVerbruggen** [comment]
Hi guys,

Which plugin version are you using? `cordova plugin list` should tell you.

**Comment by Appstractie** [comment]
nl.x-services.plugins.calendar 4.3.4 "Calendar"

**Comment by EddyVerbruggen** [comment]
Hmm, weird.. have you tried my code sample yet? And which hardware is this running on?

**Comment by Appstractie** [comment]
Yes, have tried the sample code, same results. I am running Lollipop on a Samsung Galaxy Note 3.

**Comment by EddyVerbruggen** [comment]
Running Lollipop on a Nexus 7 (2012 model) here, will try a Samsung Galaxy S5 next week. Hopefully I can reproduce the problem. In the meanwhile be my guest and debug the Android source.

**Comment by Appstractie** [comment]
I will check it out, but not sure if I have the technical baggage to find it. 
Meanwhile I am reading on the Android developer fora stories about other apps not returning recurring events since Lollipop.

**Comment by ramsabarish** [comment]
Hi I'm late to reply, the plugin version is nl.x-services.plugins.calendar 4.2.8 "Calendar" and I have tested it on Moto - g  (OS 4.4.4) as well as Samsung S3 but the result was same the recurring event were not being picked up. I guess the issue is not with the device and I will try out the latest version of the plugin.

**Comment by ramsabarish** [comment]
Hi Eddy,
              I have tried the latest version of the plugin nl.x-services.plugins.calendar 4.3.5 and still the recurring events are not being shown.

**Comment by Appstractie** [comment]
Same here. I also have tried to debug Android but simply miss the Kava knowledge to make good sense of it. 

I have tried some things but did not make any progress:

- There are known repeat event problems with the Samsung stock app as mentioned on the Android forum. When I download another app and create a recurring event, but it is still not picked up by the plugin
- It is mentioned on the Android forum that recurring events with end-date are not picked up, but I tried it both with- and without an end-date, and both events are not picked up. 
- When I add a recurring event (not picked up) and remove the recurring element from the event, the plugin picks it up right away.

No solution in sight....
